Errors in runtime.h, exposed by Embeddinator

### Steps to Reproduce

1. Clone `mono/Embeddinator-4000` commit `d697cdfca5060c07e95063c30f4864bd0a36b0ca` on a macOS 10.13.5 machine with Xcode 9.4.1 and Visual Studio as described below.
2. `cd objcgen; make`
3. Edit `samples/mac/weather/Makefile` according to the following diff:

```diff
 prepare: $(OBJC_GEN)
        $(MONOMSBUILD) /p:Configuration=Debug MyManagedStuff/MyManagedStuff.sln
-       $(MONO) $(OBJC_GEN) --debug -c MyManagedStuff/MyManagedStuff/bin/Debug/MyManagedStuff.dll -o theweather/MyManagedStuff
+       $(MONO) $(OBJC_GEN) --debug -c MyManagedStuff/MyManagedStuff/bin/Debug/MyManagedStuff.dll -o theweather/MyManagedStuff \
+               --abi=x86_64 --platform=macos-modern
        cp MyManagedStuff/MyManagedStuff/bin/Debug/MyManagedStuff.dll theweather/MyManagedStuff/MyManagedStuff.dll
 ```

4. `cd samples/mac/weather; make prepare`

### Expected Behavior

It should compile.

### Actual Behavior

Fails with below error log.

### Environment

```
=== Visual Studio Community 2017 for Mac ===

Version 7.5.3 (build 7)
Installation UUID: a880d919-4858-4032-baa6-3566143de94e
Runtime:
	Mono 5.10.1.57 (2017-12/ea8a24b1bbf) (64-bit)
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 4.4.1.178 (master / eeaeb7e6)

	Package version: 510010057

=== NuGet ===

Version: 4.3.1.4445

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.1.1
	2.1.0
	2.0.5
SDK: /usr/local/share/dotnet/sdk/2.1.301/Sdks
SDK Versions:
	2.1.301
	2.1.300
	2.1.4
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.10.1/lib/mono/msbuild/15.0/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.2
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Apple Developer Tools ===

Xcode 9.4.1 (14161)
Build 9F2000

=== Xamarin.Mac ===

Version: 4.4.1.193 (Visual Studio Community)

=== Xamarin.iOS ===

Version: 11.12.0.4 (Visual Studio Community)
Hash: 64fece5f
Branch: d15-7
Build date: 2018-05-29 20:00:44-0400

=== Xamarin.Android ===

Version: 8.3.3.2 (Visual Studio Community)
Android SDK: /Users/wjk/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		7.1 (API level 25)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 27.0.1
SDK Build Tools Version: 27.0.3

Java SDK: /usr
java version "1.8.0_172"
Java(TM) SE Runtime Environment (build 1.8.0_172-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.172-b11, mixed mode)

Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Xamarin Inspector ===

Version: 1.4.0
Hash: b3f92f9
Branch: master
Build date: Fri, 19 Jan 2018 22:00:34 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 705030007
Git revision: 13cecd02aceddf29a1ed57b86f81c02994df1483
Build date: 2018-06-14 15:48:08-04
Xamarin addins: 7065de97cf22c9038fdc39dd627f2c30790fd8af
Build lane: monodevelop-lion-d15-7

=== Operating System ===

Mac OS X 10.13.5
Darwin 17.6.0 Darwin Kernel Version 17.6.0
    Tue May  8 15:22:16 PDT 2018
    root:xnu-4570.61.1~1/RELEASE_X86_64 x86_64

=== Enabled user installed addins ===

.NET Core Extended Target Framework Support 1.0
AddinMaker 1.4.2
Internet of Things (IoT) development (Preview) 7.5
```

### Build Logs

```
/Library/Frameworks/Mono.framework/Versions/Current/Commands/msbuild /p:Configuration=Debug MyManagedStuff/MyManagedStuff.sln
Microsoft (R) Build Engine version 15.6.0.0 (xplat-master/ca830585 Sun Mar 25 19:24:09 EDT 2018) for Mono
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 7/13/2018 12:39:20 PM.
Project "/Volumes/Sources/Embeddinator-4000/samples/mac/weather/MyManagedStuff/MyManagedStuff.sln" on node 1 (default targets).
ValidateSolutionConfiguration:
  Building solution configuration "Debug|Any CPU".
Project "/Volumes/Sources/Embeddinator-4000/samples/mac/weather/MyManagedStuff/MyManagedStuff.sln" (1) is building "/Volumes/Sources/Embeddinator-4000/samples/mac/weather/MyManagedStuff/MyManagedStuff/MyManagedStuff.csproj" (2) on node 1 (default targets).
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
CoreCompile:
Skipping target "CoreCompile" because all output files are up-to-date with respect to the input files.
CopyFilesToOutputDirectory:
  MyManagedStuff -> /Volumes/Sources/Embeddinator-4000/samples/mac/weather/MyManagedStuff/MyManagedStuff/bin/Debug/MyManagedStuff.dll
Done Building Project "/Volumes/Sources/Embeddinator-4000/samples/mac/weather/MyManagedStuff/MyManagedStuff/MyManagedStuff.csproj" (default targets).
Done Building Project "/Volumes/Sources/Embeddinator-4000/samples/mac/weather/MyManagedStuff/MyManagedStuff.sln" (default targets).

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.02
/Library/Frameworks/Mono.framework/Versions/Current/Commands/mono ../../../objcgen/bin/Debug/objcgen.exe --debug -c MyManagedStuff/MyManagedStuff/bin/Debug/MyManagedStuff.dll -o theweather/MyManagedStuff \
		--abi=x86_64 --platform=macos-modern
Parsing assemblies...
	Parsed 'MyManagedStuff/MyManagedStuff/bin/Debug/MyManagedStuff.dll'
Processing assemblies...
	2 types found
Generating binding code...
	Generated: theweather/MyManagedStuff/bindings.h
	Generated: theweather/MyManagedStuff/bindings-private.h
	Generated: theweather/MyManagedStuff/bindings.m
	Generated: theweather/MyManagedStuff/bindings.xml
	Generated: theweather/MyManagedStuff/embeddinator.h
	Generated: theweather/MyManagedStuff/glib.h
	Generated: theweather/MyManagedStuff/mono_embeddinator.h
	Generated: theweather/MyManagedStuff/mono-support.h
	Generated: theweather/MyManagedStuff/objc-support.h
	Generated: theweather/MyManagedStuff/glib.c
	Generated: theweather/MyManagedStuff/mono_embeddinator.c
	Generated: theweather/MyManagedStuff/objc-support.m
	Generated: theweather/MyManagedStuff/Make.config
Compiling binding code...
Found Mono 5.10.1.57 (between 5.4.1.7 and 5.4.1.7)
Found Xamarin.Mac 4.4.1.193 (between 4.0.0.215 and 4.0.0.215)
	xcode-select -p
	xcrun clang -g -O0 -fobjc-arc -ObjC -Wall -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=10.7 -I/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/include -DXAMARIN_MAC -DMONO_EMBEDDINATOR_DLL_EXPORT  -c theweather/MyManagedStuff/glib.c -o theweather/MyManagedStuff/x86_64/glib.o 
	xcrun clang -g -O0 -fobjc-arc -ObjC -Wall -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=10.7 -I/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/include -DXAMARIN_MAC -DMONO_EMBEDDINATOR_DLL_EXPORT  -c theweather/MyManagedStuff/mono_embeddinator.c -o theweather/MyManagedStuff/x86_64/mono_embeddinator.o 
In file included from theweather/MyManagedStuff/mono_embeddinator.c:29:
In file included from theweather/MyManagedStuff/mono_embeddinator.h:32:
In file included from theweather/MyManagedStuff/mono-support.h:36:
In file included from /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/include/xamarin/xamarin.h:14:
In file included from /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/include/xamarin/runtime.h:21:
/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/include/xamarin/runtime-generated.h:86:1: error: must use 'enum' tag to refer to type 'MarshalManagedExceptionMode'
MarshalManagedExceptionMode
^
enum 
/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/include/xamarin/runtime-generated.h:89:1: error: must use 'enum' tag to refer to type 'MarshalObjectiveCExceptionMode'
MarshalObjectiveCExceptionMode
^
enum 
In file included from theweather/MyManagedStuff/mono_embeddinator.c:29:
In file included from theweather/MyManagedStuff/mono_embeddinator.h:32:
In file included from theweather/MyManagedStuff/mono-support.h:36:
In file included from /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/include/xamarin/xamarin.h:14:
/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/include/xamarin/runtime.h:88:18: error: pointer to non-const type 'const Protocol *' with no explicit ownership
        const Protocol **protocols; // the corresponding native protocols
                        ^
3 errors generated.
```

--- a/runtime/xamarin/runtime.h
+++ b/runtime/xamarin/runtime.h
@@ -86,7 +86,7 @@ typedef struct __attribute__((packed)) {
 typedef struct __attribute__((packed)) {
 	const uint32_t *protocol_tokens; // an array of token references to managed interfaces that represent protocols
 	// __unsafe_unretained needed to prevent "error: pointer to non-const type 'const Protocol *' with no explicit ownership" in Embeddinator
-	const __unsafe_unretained Protocol **protocols; // the corresponding native protocols
+	const __unsafe_unretained Protocol * const * protocols; // the corresponding native protocols
 } MTProtocolMap;
 
 struct MTRegistrationMap;

